docking.js

```diff
@@ -2018,8 +2018,14 @@ var DockManager = class DashToDock_DockManager {
             }
 
             const mainDockProperties = {};
-            if (dock === this.mainDock)
-                mainDockProperties.onComplete = callback;
+            if (dock === this.mainDock && callback !== undefined) {
+                const destroyId = dash.connect('destroy',
+                    () => mainDockProperties.onStopped(false));
+                mainDockProperties.onStopped = (finished) => {
+                    dash.disconnect(destroyId);
```

Comment:
Can just use

```suggestion
                    () => mainDockProperties.onStopped());
                mainDockProperties.onStopped = () => {
                    dash.disconnect(destroyId);
```

Since we're the one defining that function and `finished` being `undefined` is fine anyways for the way is checked everywhere.

Reply:
Half correct.

If anyone else is listening to onStopped then they expect to receive a boolean. Only we don't use it, so you could remove the `finished` but can't remove the `false`. I recommend keeping both to make it clear there's a parameter whose value should be correct even when we don't use it.

Reply:
I was thinking that, but we're definitely the only callback since we're setting it locally.

However, I'm fine if you want to keep it to remember the actual type "signature", but if you do please use the `_mainDockProperties.onStopped = _finished => {` syntax.

Reply:
That suggestion seems to contain two mistakes:
 * No parentheses (which I assume is a mitsake); and
 * The actual parameter (according to `environment.js`) is called `isFinished`. But I didn't think the name mattered much.

So maybe now I will just remove the parameter to avoid further discussion because this is a waste of our time.

Reply:
I don't think either of those underscores make sense in the existing function or in the existing coding convention (we don't underscore locals or parameters) so I'm going to ignore that and merge as-is.

Reply:
> No parentheses (which I assume is a mitsake)

No it's the expected style (see eslint, we don't complain here but it would if would be stricter as shell/gjs is in future, and we will)

> The actual parameter (according to environment.js) is called isFinished. But I didn't think the name mattered much.

Of course not

> I don't think either of those underscores make sense in the existing function or in the existing coding convention (we don't underscore locals or parameters)

We do underscore when it's not used, please there is a style we want to follow and we should in order to make this extension to be soon more compatible with what upstream linting expects us to do.

I would have merged once fixed (or I would have taken care), please it would be nice if you'd follow the suggestions without merging by yourself.